### ISSUE TITLE
Automatic releasing via Travis

### ISSUE THREAD

**Issue by ronaldtse** [rnpgp/rnp#749]
Travis has a "GitHub release" provider that allows automatic tagging on Git and the subsequent releasing.

https://docs.travis-ci.com/user/deployment/releases/

i.e. we can make Travis "release" once we the new release PR is merged.

@dewyatt @ni4 perhaps we should switch to this?

**Comment by dewyatt** [comment]
I've actually been working on CI stuff on and off for the past 3 weeks or so, and this was one of the features I looked at.
My thoughts were that it would mostly be useful to automatically build+upload RPMs from the release branch. Is that what you had in mind?

On a related note though, I completely burned out on Travis CI over the past few weeks and have been evaluating alternatives that we could potentially switch to. Travis CI is far behind other solutions and I got so tired of working around bugs and limitations. I'll create a separate issue to discuss that though.

**Comment by ronaldtse** [comment]
I was purely thinking of creating a release tarball using the Travis Github release deployer.

An RPM is already automatically built+pushed on any update at https://github.com/riboseinc/rpm-spec-rnp/ using our internal Jenkins CI. That said, we could also switch to Travis CI for this too, since it allows running Docker containers.

Is Gitlab CI something you're thinking of?

**Comment by dewyatt** [comment]
> I was purely thinking of creating a release tarball using the Travis Github release deployer.

A binary tarball in addition to the source tarballs we already have?

> An RPM is already automatically built+pushed on any update at https://github.com/riboseinc/rpm-spec-rnp/ using our internal Jenkins CI.

Right, I just figured it might make sense to look into eliminating the step of updating the spec.

> Is Gitlab CI something you're thinking of?

Yes, Gitlab CI is what makes the most sense to me (after looking at Circle CI and Codeship). Gitlab supports mirroring so we could continue using github without any changes. I'm still playing with it, but the fact that it has things like multi-project pipelines makes it look very appealing thus far.

**Comment by ronaldtse** [comment]
> A binary tarball in addition to the source tarballs we already have?
Only the source tarball.

> Right, I just figured it might make sense to look into eliminating the step of updating the spec.
Technically we could also do that via triggering a Travis build.

I'm open with using Gitlab CI, as long as it doesn't negatively affect the workflow. I'm not sure however on the pricing side of it....

**Comment by ronaldtse** [comment]
P.S. Gitlab mirroring seems to only work with higher paid plans?

**Comment by dewyatt** [comment]
@ronaldtse It's free for open-source projects, just like Travis CI. They do a poor job of advertising that, but you get all Gold features for open-source projects.

**Comment by ronaldtse** [comment]
Well then what are we waiting for? If we use Gold features for "open-source projects", are we allowed to have an organization that has a mix of private + open-source projects, where Gold only applies to "open-source projects"?

**Comment by ronaldtse** [comment]
@dewyatt wow Gitlab CI actually allows choice of a docker container, so the CentOS/Debian builds are straightforward now compared to Travis / Jenkins:

* Debian: https://about.gitlab.com/2016/10/12/automated-debian-package-build-with-gitlab-ci/
* CentOS RPM: https://gitlab.cern.ch/cloud-infrastructure/keystone/commit/b1b29af0f3a2c323e831ba7ea8790d245efa0f48

**Comment by dewyatt** [comment]
@ronaldtse Yep, you can choose any container on docker hub I believe. I already have a CI job that runs part of the rnp matrix in a centos:7 container, I've just been experimenting with caching etc.

Plus you can run Gitlab+Gitlab CI locally with two simple docker containers, which I think is pretty great.

**Comment by dewyatt** [comment]
> Well then what are we waiting for?

:+1: I will get to work on getting the CI stuff converted (I've already done a little bit for research purposes), if you can work on creating an organization, mirroring some repos (rnp and ruby-rnp would be a good start), etc.

> If we use Gold features for "open-source projects", are we allowed to have an organization that has a mix of private + open-source projects, where Gold only applies to "open-source projects"?

I would really expect the answer to be *yes*, but I can't say for certain.

**Comment by ronaldtse** [comment]
@dewyatt are you good with this? It's pretty simple to get GitLab CI working with a container. For CentOS you could use the typical `centos:7` container to build the package.

**Comment by dewyatt** [comment]
@ronaldtse Yep, I'm now waiting on some ordered parts so I'm no longer distracted fixing cars. I'll be looking at this today.

**Comment by dewyatt** [comment]
I'll be working on https://github.com/riboseinc/rpm-spec-botan2/issues/3 first since I'll need that.
Also, @ronaldtse (as I mentioned in another comment today), I made ribose-automake116 (and ribose-automake116-automake) SCL packages. Can you create a repo for that (I think it makes sense as a single repo).

**Comment by ronaldtse** [comment]
@dewyatt Could you help directly create the repo for the new package? Thanks!

**Comment by dewyatt** [comment]
@ronaldtse Ok will do (didn't realize I had permissions).

**Comment by dewyatt** [comment]
@ronaldtse I created [riboseinc/rpm-spec-automake116](https://github.com/riboseinc/rpm-spec-automake116). I'm thinking maybe you'll want to get it added to [riboseinc/yum](https://github.com/riboseinc/yum) for easier use though.

**Comment by ribose-jeffreylau** [comment]
It's [done](https://github.com/riboseinc/yum/commit/97b6a65b3d2b905e2c60bb663c366bbc327504b7).

**Comment by dewyatt** [comment]
Thanks @ribose-jeffreylau!

**Comment by maxirmx** [comment]
Closing it since Travis is not used anymore
